pkgs/development/python-modules/sniffio/default.nix

```diff
@@ -1,27 +1,22 @@
-{ buildPythonPackage, lib, fetchPypi, glibcLocales, isPy3k, contextvars
-, pythonOlder, pytest, curio
+{ buildPythonPackage, lib, fetchPypi, glibcLocales, isPy3k
+, pythonOlder, pytestCheckHook, curio
 }:
 
 buildPythonPackage rec {
   pname = "sniffio";
   version = "1.3.0";
+  format = "setuptools";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "sha256-5gMFxeXTFPU4klm38iqqM9j33uSXYxGSNK83VcVbkQE=";
+    hash = "sha256-5gMFxeXTFPU4klm38iqqM9j33uSXYxGSNK83VcVbkQE=";
   };
 
   disabled = !isPy3k;
 
   buildInputs = [ glibcLocales ];
 
-  propagatedBuildInputs = lib.optionals (pythonOlder "3.7") [ contextvars ];
-
-  checkInputs = [ pytest curio ];
-
-  checkPhase = ''
-    pytest
-  '';
```

Comment:
In this commit are a few unrelated changes which could maybe be moved to their own commits.